### ISSUE TITLE
Clone is only implemented for the default hasher

### ISSUE THREAD

**Issue by Fuuzetsu** [jeromefroe/lru-rs#198]
If you have a for exmaple `LruCache<K, V, fxhash::FxBuildHasher>` then this is no longer `Clone`. I think this probably just an oversight in `Clone` impl which is only over `LruCache<K, V>`.

**Comment by jeromefroe** [comment]
@Fuuzetsu you're right, this was just an oversight. I don't think I'll have time to get to it soon, but happy to review any PR's if you're able to take a stab at making the change!
